Add API version checks to all connections to Ankaios (#380)
eclipse-ankaios/ankaios: tests/resources/image/README.md

# Control Interface Tester workload

> The Control Interface Tester workload is used in system test to verify the correct execution of the Ankaios Control Interface.
> The workload reads a commands file, executes the commands and writes an output file containing the result of the commands.

Reviewer: ```suggestion
The Control Interface Tester workload is used in system tests to verify the correct execution of the Ankaios Control Interface.
The workload reads commands from a file, executes them, and writes an output file containing the results of the commands.
```

Author: ```suggestion
The Control Interface Tester workload is used in system tests to verify the correct execution of the Ankaios Control Interface.
The workload reads commands from a file, executes them and writes an output file containing the results of the commands.
```

> In case a Control Interface instance was not provided to the workload, a `NoAPI` result is written. If the Control Interface was preliminary closed by Ankaios, e.g., due to a protocol error, a `ConnectionClosed` result is provided.

Reviewer: ```suggestion
If no Control Interface instance was provided to the workload, a `NoAPI` result is written. If the Control Interface was preliminary closed by Ankaios, e.g., due to a protocol error, a `ConnectionClosed` result is provided.
```

Author: We are at the matter of style here, but when you insist 😉

Reviewer: I forgot to mention it as optional. I did not know the "In case a", I thought it is always "In case of ...". And then I checked this with write improvement tool and it suggested just "If no...".

> ## Building ans pushing a new image

Reviewer: ```suggestion
## Building and pushing a new image
```

It is planned to automate the process of building and pushing a new version of the container, but for now the process is done manually.

To build a new image run the following command from the project root folder:

```bash

> podman build -t ghcr.io/eclipse-ankaios/control_interface_tester:manual-build-<new version number> . -f tests/resources/image/Dockerfile

Reviewer: Just for curiosity, do we need this as a multi-platform build?

Author: I also wondered, but for now it runs only on the amd64 machines. As this is the second time we needed to update the image, I just documented the current state as it was given to me.

```

To push the new image to GitHub container registry, you will need to generate an access token that is allowed to write packages and login to `ghcr.io`:

```bash
podman login ghcr.io
```

> Afterwards the push can be done with:

Reviewer: ```suggestion
Afterwards the new image can be published with:
```
